File: library/src/main/java/me/shouheng/compress/strategy/Strategies.java
package me.shouheng.compress.strategy;

import me.shouheng.compress.strategy.compress.Compressor;
import me.shouheng.compress.strategy.luban.Luban;

/**
 * The provided strategies.
 *
 * @author WngShhng
 */
public final class Strategies {

    private Strategies() {
        throw new UnsupportedOperationException("u can't initialize me");
    }

    /**
     * The luban compress strategy
     *
     * @return luban strategy
     */
    public static Luban luban() {
        return new Luban();
    }

    /**
     * The compressor compress strategy
     *
     * @return compressor strategy
     */
    public static Compressor compressor() {
        return new Compressor();
    }

}

File: library/src/main/java/me/shouheng/compress/strategy/Strategies.kt
package me.shouheng.compress.strategy

import me.shouheng.compress.strategy.compress.Compressor
import me.shouheng.compress.strategy.luban.Luban

/**
 * The provided strategies.
 *
 * @author Shouheng Wang
 */
object Strategies {

    /** The luban compress strategy, used to get the image for preview.
     * See https://github.com/Curzibn/Luban for details. */
    fun luban(): Luban = Luban()

    /** The compressor compress strategy, used to specify the image width height constraints. */
    fun compressor(): Compressor = Compressor()
}
